src/pages/configurePatch/configurePatchCore/ConfigureBuildVariants.tsx

```diff
@@ -23,6 +23,31 @@ export const ConfigureBuildVariants: React.FC<Props> = ({
   selectedBuildVariant,
   setSelectedBuildVariant,
 }) => {
+  const [state, dispatch] = useReducer(reducer, { numButtonsPressed: 0 });
+  const keyDownCb = useMemo(
+    () => (e: KeyboardEvent) => {
+      if (hotKeys.has(e.key)) {
+        dispatch({ type: "increment" });
+      }
+    },
+    [dispatch]
+  );
+  const keyUpCb = useMemo(
+    () => (e: KeyboardEvent) => {
+      if (hotKeys.has(e.key)) {
+        dispatch({ type: "decrement" });
+      }
+    },
+    [dispatch]
+  );
+  useEffect(() => {
+    window.addEventListener("keydown", keyDownCb);
+    window.addEventListener("keyup", keyUpCb);
+    return () => {
+      window.removeEventListener("keyDown", keyDownCb);
+      window.removeEventListener("keyup", keyUpCb);
+    };
+  }, [keyUpCb, keyDownCb]);
```

Comment:
I still think this use reducer is a little overkill. Using a useState would be simpler and just as effective.

```suggestion
 const [isHotKeyPressed, setIsHotKeyPressed] = useState(false);

  useEffect(() => {
    const keyDownCb = (e: KeyboardEvent) => {
      if (hotKeys.has(e.key)) {
        setIsHotKeyPressed(true);
      }
    };
    const keyUpCb = (e: KeyboardEvent) => {
      if (hotKeys.has(e.key)) {
        setIsHotKeyPressed(false);
      }
    };

    window.addEventListener("keydown", keyDownCb);
    window.addEventListener("keyup", keyUpCb);
    return () => {
      window.removeEventListener("keyDown", keyDownCb);
      window.removeEventListener("keyup", keyUpCb);
    };
  });
```

Reply:
we have to increment a decrement a number because it is possible the user is pressing multiple hotkeys at once. holding shift, then holding command, then releasing shift will result in `isHotKeyPressed` to be false

Reply:
@khelif96 I'm going to keep the reducer as is. it's possible that we will need to implement a counter state in another component. In that case, the useReducer implementation is readily typed and reusable